prov/verbs/src/ep_rdm/verbs_queuing.h

```diff
@@ -166,4 +236,53 @@ fi_ibv_rdm_tagged_remove_from_postponed_queue(
 	}
 }
 
+static inline struct fi_ibv_rdm_tagged_request *
+fi_ibv_rdm_take_first_from_postponed_queue()
+{
+	if (!dlist_empty(&fi_ibv_rdm_tagged_send_postponed_queue)) {
+		struct fi_ibv_rdm_tagged_postponed_entry *entry = 
+			container_of(fi_ibv_rdm_tagged_send_postponed_queue.next,
+				struct fi_ibv_rdm_tagged_postponed_entry, queue_entry);
+
+		if (!dlist_empty(&entry->conn->postponed_requests_head)) {
+			struct dlist_entry *req_entry = 
+				entry->conn->postponed_requests_head.next;
+
+			struct fi_ibv_rdm_tagged_request *request =
+			    container_of(req_entry, struct fi_ibv_rdm_tagged_request,
+					 queue_entry);
+			fi_ibv_rdm_tagged_remove_from_postponed_queue(request);
+			return request;
+		}
+	}
+
+	return NULL;
+}
+
+static inline struct fi_ibv_rdm_tagged_request *
+fi_ibv_rdm_take_first_match_from_postponed_queue(dlist_func_t *match, const void *arg)
+{
+	struct dlist_entry *i, *j;
+	dlist_foreach((&fi_ibv_rdm_tagged_send_postponed_queue), i) {
+		 struct fi_ibv_rdm_tagged_postponed_entry *entry = 
+			container_of(i, struct fi_ibv_rdm_tagged_postponed_entry,
+				     queue_entry);
+
+		j = dlist_find_first_match((&entry->conn->postponed_requests_head),
+					   match, arg);
+		if (j) {
+			struct fi_ibv_rdm_tagged_request *request =
+				container_of(j, struct fi_ibv_rdm_tagged_request,
+					     queue_entry);
+		
+			fi_ibv_rdm_tagged_remove_from_postponed_queue(request);
+			return request;
+		}
+	}
+
+	return NULL;
+}
```

Comment:
Please be wary of defining everything as a static inline in a header file.  It results in code bloat that can decrease performance because of the significant increase in code size.